ext/bg/js/backend.js

```diff
@@ -565,23 +597,48 @@ class Backend {
     // Command handlers
 
     async _onCommandSearch(params) {
-        const url = chrome.runtime.getURL('/bg/search.html');
-        if (!(params && params.newTab)) {
-            try {
-                const tab = await Backend._findTab(1000, (url2) => (
-                    url2 !== null &&
-                    url2.startsWith(url) &&
-                    (url2.length === url.length || url2[url.length] === '?' || url2[url.length] === '#')
-                ));
-                if (tab !== null) {
-                    await Backend._focusTab(tab);
-                    return;
+        const {mode, query} = params || {mode: 'sameTab'};
+
+        const options = await this.getOptions(this.optionsContext);
+        const {popupWidth, popupHeight} = options.general;
+
+        const baseUrl = chrome.runtime.getURL('/bg/search.html');
+        const queryParams = {mode};
+        if (query && query.length > 0) { queryParams.query = query; }
+        const queryString = new URLSearchParams(queryParams).toString();
+        const url = `${baseUrl}?${queryString}`;
+
+        switch (mode) {
+            case 'sameTab':
+                try {
+                    const tab = await Backend._findTab(1000, (url2) => (
+                        url2 !== null &&
+                        url2.startsWith(url) &&
+                        (url2.length === url.length || url2[url.length] === '?' || url2[url.length] === '#')
+                    ));
+                    if (tab !== null) {
+                        await Backend._focusTab(tab);
+                        return;
+                    }
+                } catch (e) {
+                    // NOP
                 }
-            } catch (e) {
-                // NOP
-            }
+                chrome.tabs.create({url});
+                return;
+            case 'newTab':
+                chrome.tabs.create({url});
+                return;
+            case 'popup':
+                if (this.popupWindow !== null) {
+                    const callback = () => this.checkLastError(chrome.runtime.lastError);
+                    chrome.windows.remove(this.popupWindow.id, callback);
+                }
+                this.popupWindow = await new Promise((resolve) => chrome.windows.create(
```

Comment:
Not necessary to add anything until we encounter something, but keep an eye out in case you ever see a `runtime.lastError` here. I've seen cases where `runtime.lastError` will be set despite the docs not listing anything, particularly on Firefox for Android. (I know this feature shouldn't work on Android, but could happen for strange edge cases also.)